keystrokes/src/components/Character/CharacterList.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/character.css';
+import styleGuide from '../../stylesheets/style';
+import { ref } from '../../config/constants';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+
+function CharacterList(props) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState()
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		console.log(props)
+		setCharacters(props.userInfo.characters || {});
+		setCurrentCharacter(props.userInfo.currentCharacter);
+	}, [props]);
+
+	const handleOpen = () => {
+		setOpen(true);
+	}
+
+	async function handleSubmit(fields) {
+		const {name, characterClass, spec, race} = fields
+		let charactersSnapShot = await ref.child("/Characters");
+		let doesCharacterExist = await charactersSnapShot.child(name).once('value');
+		if (doesCharacterExist.exists()) {
+			console.log("error")
+			//throw error since name is currently id - replace with real key eventually
+			return;
+		} else {
+			charactersSnapShot
+			.child(name).set({
+				"class": characterClass, 
+				"race": race, 
+				"name": name, 
+				"specs": {}
+			});
+			charactersSnapShot
+			.child(name)
+			.child("specs")
+			.child(spec)
+			.set({
+				"spec": spec, 
+				"selected": true,
+				"configured": false
+			});
+		}
+		let snapshot = await ref.child(props.userPath);
+		let userCharacterData = await snapshot.once('value');
+		if (userCharacterData.exists()) {
+			snapshot
+				.child('characters').child(name) //name should probably be a real key
+				.set({ "name": name });
+		} else {
+			snapshot.set({
+					"currentCharacter": name,
+					"characters": {}
+			});
+			snapshot
+				.child('characters').child(name)
+				.set({ "name": name });
+		}
+		setCharacters({...characters, name: {"name": name}}) //immutabilityHelper here
+	}
+
+	return (
+		<React.Fragment>
+			{
+				open ?
+					<React.Fragment>
+						<CharacterCreationModal 
+							isOpen={open}
+							setIsOpen={setOpen}
+							handleSubmit={handleSubmit}
+						/> 
+					</React.Fragment>: 
+				null
+			}
+			<Typography variant="h1" align="center">Character List</Typography>
+			<Typography variant="h3" align="center" color="textSecondary" gutterBottom>Edit a character from the list or create a new character.</Typography>
```

Comment:
after about 3-4 props i tend to put all of them on their own lines for readability.
Another benefit is you can comment out or change each individually and the diff is a lot easier to parse.